evergreen/eap.yml

```diff
@@ -0,0 +1,360 @@
+    - command: s3.get
+      params:
+        aws_key: ${aws_key}
+        aws_secret: ${aws_secret}
+        remote_file: mongosql-odbc-driver/artifacts/${version_id}/ubuntu2204/release/${UBUNTU_FILENAME}
+        local_file: mongosql-odbc-driver/target/release/${UBUNTU_FILENAME}
+        bucket: mciuploads
+        permissions: public-read
+        content_type: application/octet-stream
+    - command: s3.put
+      params:
+        aws_key: ${release_aws_key}
+        aws_secret: ${release_aws_secret}
+        local_file: mongosql-odbc-driver/target/release/${UBUNTU_FILENAME}
+        remote_file: ${UBUNTU2204_INSTALLER_PATH}
+        bucket: translators-connectors-releases
+        permissions: public-read
+        content_type: application/octet-stream
+    - command: s3.get
+      params:
+        aws_key: ${aws_key}
+        aws_secret: ${aws_secret}
+        remote_file: mongosql-odbc-driver/artifacts/${version_id}/ubuntu2204/release/${UBUNTU_FILENAME}.sig
+        local_file: mongosql-odbc-driver/target/release/${UBUNTU_FILENAME}.sig
+        bucket: mciuploads
+        permissions: public-read
+        content_type: application/octet-stream
+    - command: s3.put
+      params:
+        aws_key: ${release_aws_key}
+        aws_secret: ${release_aws_secret}
+        local_file: mongosql-odbc-driver/target/release/${UBUNTU_FILENAME}.sig
+        remote_file: eap/mongosql-odbc-driver/ubuntu2204/${RELEASE_VERSION}/release/${UBUNTU_FILENAME}.sig
+        bucket: translators-connectors-releases
+        permissions: public-read
+        content_type: application/octet-stream
+    - command: s3.get
+      params:
+        aws_key: ${aws_key}
+        aws_secret: ${aws_secret}
+        remote_file: mongosql-odbc-driver/artifacts/${version_id}/ubuntu2204/release/libatsql.so
+        local_file: mongosql-odbc-driver/target/release/libatsql.so
+        bucket: mciuploads
+        permissions: public-read
+        content_type: application/octet-stream
+    - command: s3.put
+      params:
+        aws_key: ${release_aws_key}
+        aws_secret: ${release_aws_secret}
+        local_file: mongosql-odbc-driver/target/release/libatsql.so
+        remote_file: eap/mongosql-odbc-driver/ubuntu2204/${RELEASE_VERSION}/release/libatsql.so
+        bucket: translators-connectors-releases
+        permissions: public-read
+        content_type: application/octet-stream
+
+tasks:
+  - name: snapshot-eap
+    depends_on:
+      - name: compile-eap
+        variant: "*"
+      - name: clippy
+        variant: "*"
+      - name: rustfmt
+        variant: "*"
+      - name: unit-test-eap
+        variant: ".release-variant"
+      - name: integration-test-eap
+        variant: ".release-variant"
+      - name: result-set-test-eap
+        variant: ".release-variant"
+      - name: sign-eap
+        variant: ".release-variant"
+    commands:
+      - func: "upload eap release"
+
+  - name: compile-eap
+    commands:
+      - func: "generate expansions"
```

Comment:
Cool that's a good workaround for setting the values in the generate expansion.  So it'll run it once for the `pre` task but we want to set snapshot-eap to true so this will update the expansions with that variable set the way we want.  With it set, do we still need to set `vars: snapshot-eap: "true"` in the commands after this?  It's fine that it's there, but I would have expected it to already be set here.

Reply:
Yeah I removed the vars invocation from the other function calls.